sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -980,9 +981,16 @@ valueExpression
     | left=valueExpression operator=AMPERSAND right=valueExpression                          #arithmeticBinary
     | left=valueExpression operator=HAT right=valueExpression                                #arithmeticBinary
     | left=valueExpression operator=PIPE right=valueExpression                               #arithmeticBinary
+    | left=valueExpression shiftOperator right=valueExpression                              #shiftExpression
     | left=valueExpression comparisonOperator right=valueExpression                          #comparison
     ;
 
+shiftOperator
+    : LT LT
+    | GT GT
+    | GT GT GT
```

Comment:
This syntax is wrong because we should not allow spaces between `LT` and `GT`.

For example, `< <` is a syntax error in other DBMS.
```
postgres=# SELECT 2 < < 1;
ERROR:  syntax error at or near "<"
LINE 1: SELECT 2 < < 1;
```

But, this PR allows the following.
```
scala> sql("SELECT 2 < < 1").show()
+--------+
|(2 << 1)|
+--------+
|       4|
+--------+
```

Reply:
I haven't found a way to retrieve the whitespace back to fail such a case. Tokenize these operators works but it fails to parse nested map and struct types

Reply:
This rule has been refined and whitespace-discrete operators now result in error as expected